Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +229,93 @@ namespace AzFramework
         }
     }
 
+    void XcbNativeWindow::DrawSplash()
+    {
+
+        int width = 0;
+        int height = 0;
+
+        auto get_image_data_static = [](int &width, int &height) -> uint8_t*
+        {
+            width = O3DE_LOGO::width;
+            height = O3DE_LOGO::height;
+
+            uint8_t *logo_binary_data = (uint8_t*)malloc(width * height * 4);   // so xcb_image_destroy can free it
+            
+            for (int i = 0; i < width * height; i++)
+            {
+                logo_binary_data[i * 4 + 0] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][0];
+                logo_binary_data[i * 4 + 1] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][1];
+                logo_binary_data[i * 4 + 2] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][2];
+                logo_binary_data[i * 4 + 3] = 0xff;
+            }
+            return logo_binary_data;
+        };
+
+        uint8_t* data = get_image_data_static(width, height);
+
+        xcb_void_cookie_t cookie;
+        xcb_pixmap_t pixmap = xcb_generate_id(m_xcbConnection);
+        cookie = xcb_create_pixmap_checked(m_xcbConnection, m_xcbRootScreen->root_depth, pixmap, m_xcbWindow, width, height);
+        if (xcb_generic_error_t* error = xcb_request_check(m_xcbConnection, cookie))
+        {
+            AZ_Printf("XCB window init", "Error in xcb_create_pixmap: %d\n", error->error_code);
+            free(error);
+            free(data);
+            return;
+        }
+        // This uses hardcoded 8bit depth , TODO check on HDR monitor
+        xcb_image_t* image =
+            xcb_image_create_native(m_xcbConnection, width, height, XCB_IMAGE_FORMAT_Z_PIXMAP, 24, data, width * height * 4, data);
+        xcb_image_put(m_xcbConnection, pixmap, m_xcbGraphicContext, image, 0, 0, 0);
+        xcb_image_destroy(image); // also frees data
+        xcb_flush(m_xcbConnection);
+        
+        xcb_generic_event_t* event;
+        bool drawn = false;
+        while (!drawn)
+        {
+            event = xcb_wait_for_event(m_xcbConnection);
+            switch (event->response_type & ~0x80)
+            {
+            case XCB_EXPOSE: {
+                xcb_expose_event_t* x = (xcb_expose_event_t*)event;
+                xcb_copy_area(
+                    m_xcbConnection,
+                    pixmap,
+                    m_xcbWindow,
+                    m_xcbGraphicContext,
+                    x->x,
+                    x->y,
+                    (m_width - width) / 2,
+                    (m_height - height) / 2,
+                    x->width,
+                    x->height);
+                xcb_flush(m_xcbConnection);
+                drawn = true;
+                break;
+            }
+            break;
+            default:
+                break;
+            }
+            free(event);
+        }
+
+        xcb_free_pixmap(m_xcbConnection, pixmap);
+    }
```

Comment:
Hi, no it's not missing, the xcb_image_destroy is responsible for freeing the data vector, also this library is not aware of AZStd, so I wanted to use standard malloc to not have memory leak, above I'm using free, because the "ownership" was not yet transferred to xcb